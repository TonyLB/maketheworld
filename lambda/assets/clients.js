import { S3Client, GetObjectCommand as GetObjectCommandOriginal } from "@aws-sdk/client-s3"
import { EventBridgeClient } from "@aws-sdk/client-eventbridge"

export const ebClient = new EventBridgeClient({ region: process.env.AWS_REGION })

// export const s3Client = new S3Client(params)

export class GetObjectCommand extends GetObjectCommandOriginal {}
