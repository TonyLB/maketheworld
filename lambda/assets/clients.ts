import { GetObjectCommand as GetObjectCommandOriginal } from "@aws-sdk/client-s3"
import { EventBridgeClient, PutEventsCommand } from "@aws-sdk/client-eventbridge"
import { apiClient as apiClientImport } from "@tonylb/mtw-utilities/dist/apiManagement/apiManagementClient"
import { SFNClient } from "@aws-sdk/client-sfn"

export const ebClient = new EventBridgeClient({ region: process.env.AWS_REGION })
export const sfnClient = new SFNClient({ region: process.env.AWS_REGION })

// export const s3Client = new S3Client(params)

export class GetObjectCommand extends GetObjectCommandOriginal {}

export const apiClient = {
    send: (message: any) => {
        return apiClientImport.send(message)
            .catch(async (err: any) => {
                if (err.name === 'GoneException' || err.name === 'BadRequestException') {
                    await ebClient.send(new PutEventsCommand({
                        Entries: [{
                            EventBusName: process.env.EVENT_BUS_NAME,
                            Source: 'mtw.diagnostics',
                            DetailType: 'Force Disconnect',
                            Detail: JSON.stringify({
                                connectionId: message.ConnectionId
                            })
                        }]
                    }))
                }
                else {
                    console.log(`Error: ${err.name}`)
                    throw err
                }
            })
    }
}
