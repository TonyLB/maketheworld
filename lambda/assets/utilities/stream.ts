import { SelectObjectContentEventStream } from "@aws-sdk/client-s3"

export const streamToString = (stream: any): Promise<string> => (
    new Promise((resolve, reject) => {
      const chunks: any[] = []
      stream.on("data", (chunk) => chunks.push(chunk))
      stream.on("error", reject)
      stream.on("end", () => resolve(Buffer.concat(chunks).toString("utf8")))
    })
)

export const streamToBuffer = (stream) => (
  new Promise((resolve, reject) => {
    const chunks: any[] = []
    stream.on("data", (chunk) => chunks.push(chunk))
    stream.on("error", reject)
    stream.on("end", () => resolve(Buffer.concat(chunks)))
  })
)

export const convertSelectDataToJson = async (generator: AsyncIterable<SelectObjectContentEventStream>) => {
    const chunks: Uint8Array[] = []
    for await (const value of generator) {
        if (value.Records) {
            if (value.Records.Payload) {
              chunks.push(value.Records.Payload)
            }
        }
    }
    let payload = Buffer.concat(chunks).toString('utf8')
    payload = payload.replace(/,$/, '')
    return JSON.parse(payload)
}
