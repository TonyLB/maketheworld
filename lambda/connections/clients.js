import { CognitoIdentityProviderClient } from "@aws-sdk/client-cognito-identity-provider"
import { EventBridgeClient } from "@aws-sdk/client-eventbridge"

export const ebClient = new EventBridgeClient({ region: process.env.AWS_REGION })
export const cognitoClient = new CognitoIdentityProviderClient({ region: process.env.AWS_REGION })
