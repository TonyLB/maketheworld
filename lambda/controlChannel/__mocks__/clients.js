const mockSend = jest.fn().mockResolvedValue({})

export const s3Client = {
    send: mockSend
}

export class GetObjectCommand {
    constructor() {}
}
