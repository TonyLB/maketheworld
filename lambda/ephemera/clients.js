import { EventBridgeClient } from "@aws-sdk/client-eventbridge"

export const ebClient = new EventBridgeClient({ region: process.env.AWS_REGION })
