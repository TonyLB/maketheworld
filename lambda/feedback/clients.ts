import { EventBridgeClient, PutEventsCommand } from "@aws-sdk/client-eventbridge"
import { apiClient as apiClientImport } from "@tonylb/mtw-utilities/ts/apiManagement/apiManagementClient"
import { SFNClient } from "@aws-sdk/client-sfn"

export const ebClient = new EventBridgeClient({ region: process.env.AWS_REGION })
export const sfnClient = new SFNClient({ region: process.env.AWS_REGION })

export const apiClient = {
    send: (message: any) => {
        return apiClientImport.send(message)
            .catch(async (err: any) => {
                if (err.name === 'GoneException' || err.name === 'BadRequestException') {
                    await ebClient.send(new PutEventsCommand({
                        Entries: [{
                            EventBusName: process.env.EVENT_BUS_NAME,
                            Source: 'mtw.diagnostics',
                            DetailType: 'Force Disconnect',
                            Detail: JSON.stringify({
                                connectionId: message.ConnectionId
                            })
                        }]
                    }))
                }
                else {
                    console.log(`Error: ${err.name}`)
                    throw err
                }
            })
    }
}
