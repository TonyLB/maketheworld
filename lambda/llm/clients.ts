import { SNSClient } from "@aws-sdk/client-sns"
import AWSXRay from 'aws-xray-sdk'

export const snsClient = AWSXRay.captureAWSv3Client(new SNSClient({ region: process.env.AWS_REGION }))
