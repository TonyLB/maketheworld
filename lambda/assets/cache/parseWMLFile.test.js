import { jest, expect } from '@jest/globals'

jest.mock('../clients.js')
import { streamToString } from '/opt/utilities/stream.js'

import {
    parseWMLFile
} from './parseWMLFile.js'
import { s3Client } from '../clients.js'

describe('parseWMLFile', () => {
    beforeEach(() => {
        jest.clearAllMocks()
        jest.resetAllMocks()
    })
    it('should return parsed output', async () => {
        s3Client.send.mockResolvedValue({})
        streamToString.mockResolvedValue(`
            <Asset key=(test) fileName="test">
                <Room key=(ABC)>
                    <Name>Vortex</Name>
                </Room>
                <Condition if={active}>
                    <Depend on=(active) />
                    <Room key=(ABC)>
                        The lights are on
                    </Room>
                </Condition>
                <Variable key=(powered) default={false} />
                <Variable key=(switchedOn) default={true} />
                <Computed key=(active) src={powered && switchedOn}>
                    <Depend on=(switchedOn) />
                    <Depend on=(powered) />
                </Computed>
                <Action key=(toggleSwitch) src={switchedOn = !switchedOn} />
            </Asset>
        `)
        const parseOutput = await parseWMLFile('test')
        const topLevelAppearance = {
            contextStack: [{ key: 'test', tag: 'Asset', index: 0}],
            contents: [],
            errors: [],
            props: {}
        }
        expect(parseOutput).toEqual({
            test: {
                key: 'test',
                tag: 'Asset',
                fileName: 'test',
                importMap: {},
                appearances: [{
                    contextStack: [],
                    errors: [],
                    props: {},
                    contents: [{
                        key: 'ABC',
                        tag: 'Room',
                        index: 0
                    },
                    {
                        key: 'Condition-0',
                        tag: 'Condition',
                        index: 0
                    },
                    {
                        key: 'powered',
                        tag: 'Variable',
                        index: 0
                    },
                    {
                        key: 'switchedOn',
                        tag: 'Variable',
                        index: 0
                    },
                    {
                        key: 'active',
                        tag: 'Computed',
                        index: 0
                    },
                    {
                        key: 'toggleSwitch',
                        tag: 'Action',
                        index: 0
                    }]
                }]
            },
            ABC: {
                key: 'ABC',
                tag: 'Room',
                appearances: [{
                    ...topLevelAppearance,
                    global: false,
                    name: 'Vortex',
                    render: []
                },
                {
                    contextStack: [{ key: 'test', tag: 'Asset', index: 0 }, { key: 'Condition-0', tag: 'Condition', index: 0 }],
                    errors: [],
                    global: false,
                    props: {},
                    render: ['The lights are on '],
                    contents: []
                }]
            },
            powered: {
                key: 'powered',
                tag: 'Variable',
                default: 'false',
                appearances: [topLevelAppearance]
            },
            switchedOn: {
                key: 'switchedOn',
                tag: 'Variable',
                default: 'true',
                appearances: [topLevelAppearance]
            },
            active: {
                key: 'active',
                tag: 'Computed',
                src: 'powered && switchedOn',
                dependencies: ['switchedOn', 'powered'],
                appearances: [topLevelAppearance]
            },
            toggleSwitch: {
                key: 'toggleSwitch',
                tag: 'Action',
                src: 'switchedOn = !switchedOn',
                appearances: [topLevelAppearance]
            },
            ['Condition-0']: {
                key: 'Condition-0',
                tag: 'Condition',
                if: 'active',
                dependencies: ['active'],
                appearances: [{
                    ...topLevelAppearance,
                    contents: [{
                        key: 'ABC',
                        tag: 'Room',
                        index: 1
                    }]
                }]
            }
        })
    })
})