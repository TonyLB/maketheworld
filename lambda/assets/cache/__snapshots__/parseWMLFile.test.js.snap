// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseWMLFile should return parsed output 1`] = `
Object {
  "ABC": Object {
    "appearances": Array [
      Object {
        "contents": Array [],
        "contextStack": Array [
          Object {
            "index": 0,
            "key": "test",
            "tag": "Asset",
          },
        ],
        "display": undefined,
        "errors": Array [],
        "global": false,
        "location": Array [
          0,
          0,
        ],
        "name": "Vortex",
        "props": Object {},
        "render": Array [],
      },
      Object {
        "contents": Array [],
        "contextStack": Array [
          Object {
            "index": 0,
            "key": "test",
            "tag": "Asset",
          },
          Object {
            "index": 0,
            "key": "Condition-0",
            "tag": "Condition",
          },
        ],
        "display": undefined,
        "errors": Array [],
        "global": false,
        "location": Array [
          0,
          1,
          0,
        ],
        "props": Object {},
        "render": Array [
          Object {
            "spaceAfter": false,
            "spaceBefore": false,
            "tag": "String",
            "value": "The lights are on",
          },
        ],
      },
    ],
    "key": "ABC",
    "tag": "Room",
  },
  "Condition-0": Object {
    "appearances": Array [
      Object {
        "contents": Array [
          Object {
            "index": 1,
            "key": "ABC",
            "tag": "Room",
          },
        ],
        "contextStack": Array [
          Object {
            "index": 0,
            "key": "test",
            "tag": "Asset",
          },
        ],
        "errors": Array [],
        "location": Array [
          0,
          1,
        ],
        "props": Object {},
      },
    ],
    "dependencies": Array [
      "active",
    ],
    "if": "active",
    "key": "Condition-0",
    "tag": "Condition",
  },
  "active": Object {
    "appearances": Array [
      Object {
        "contents": Array [],
        "contextStack": Array [
          Object {
            "index": 0,
            "key": "test",
            "tag": "Asset",
          },
        ],
        "errors": Array [],
        "location": Array [
          0,
          4,
        ],
        "props": Object {},
      },
    ],
    "dependencies": Array [
      "switchedOn",
      "powered",
    ],
    "key": "active",
    "src": "powered && switchedOn",
    "tag": "Computed",
  },
  "powered": Object {
    "appearances": Array [
      Object {
        "contents": Array [],
        "contextStack": Array [
          Object {
            "index": 0,
            "key": "test",
            "tag": "Asset",
          },
        ],
        "errors": Array [],
        "location": Array [
          0,
          2,
        ],
        "props": Object {},
      },
    ],
    "default": "false",
    "key": "powered",
    "tag": "Variable",
  },
  "switchedOn": Object {
    "appearances": Array [
      Object {
        "contents": Array [],
        "contextStack": Array [
          Object {
            "index": 0,
            "key": "test",
            "tag": "Asset",
          },
        ],
        "errors": Array [],
        "location": Array [
          0,
          3,
        ],
        "props": Object {},
      },
    ],
    "default": "true",
    "key": "switchedOn",
    "tag": "Variable",
  },
  "test": Object {
    "appearances": Array [
      Object {
        "contents": Array [
          Object {
            "index": 0,
            "key": "ABC",
            "tag": "Room",
          },
          Object {
            "index": 0,
            "key": "Condition-0",
            "tag": "Condition",
          },
          Object {
            "index": 0,
            "key": "powered",
            "tag": "Variable",
          },
          Object {
            "index": 0,
            "key": "switchedOn",
            "tag": "Variable",
          },
          Object {
            "index": 0,
            "key": "active",
            "tag": "Computed",
          },
          Object {
            "index": 0,
            "key": "toggleSwitch",
            "tag": "Action",
          },
        ],
        "contextStack": Array [],
        "description": undefined,
        "errors": Array [],
        "location": Array [
          0,
        ],
        "props": Object {},
      },
    ],
    "fileName": "test",
    "key": "test",
    "player": undefined,
    "tag": "Asset",
    "zone": undefined,
  },
  "toggleSwitch": Object {
    "appearances": Array [
      Object {
        "contents": Array [],
        "contextStack": Array [
          Object {
            "index": 0,
            "key": "test",
            "tag": "Asset",
          },
        ],
        "errors": Array [],
        "location": Array [
          0,
          5,
        ],
        "props": Object {},
      },
    ],
    "key": "toggleSwitch",
    "src": "switchedOn = !switchedOn",
    "tag": "Action",
  },
}
`;
